### ISSUE TITLE
Is setup.py  necessary?

### ISSUE THREAD

**Issue by telatin** [ewels/rich-codex#22]
In a non-python repository, I found it necessary to have a `setup.py` file, and its removal was sufficient to break rich-codex:

https://github.com/telatin/seqfu2/commit/88d5d84a6e56d7c6f2e4e76433fa8637683f1700

[and thx again for the great tool]

**Comment by ewels** [comment]
Wait, you needed to add a `setup.py` file to your repo which doesn't have any python in it? That makes no sense. Can you point me to where it broke rich-codex?

**Comment by ewels** [comment]
Found it: https://github.com/telatin/seqfu2/runs/7248369715?check_suite_focus=true

```
Run actions/setup-python@v3
  with:
    python-version: 3.x
    cache: pip
    cache-dependency-path: setup.py
    token: ***
  env:
    PATH: /home/runner/.nimble/bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/home/runner/.local/bin:/opt/pipx_bin:/home/runner/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

Successfully setup CPython (3.10.5)

Error: No file in /home/runner/work/seqfu2/seqfu2 matched to [setup.py], make sure you have checked out the target repository
```

**Comment by telatin** [comment]
I added it yesterday in my bus-tour when this popped up in the github-action log:
```

0s
Run ewels/rich-codex@v1
Run actions/setup-python@v3
Successfully setup CPython (3.10.5)
Error: No file in /home/runner/work/seqfu2/seqfu2 matched to [setup.py], make sure you have checked out the target repository
Run actions/upload-artifact@v3
Warning: No files were found with the provided path: rich_codex_*.log. No artifacts will be uploaded.
```

today just removing setup.py resulted in the same error

**Comment by ewels** [comment]
ohhh I see it: `cache-dependency-path: setup.py` - ok that's an easy fix. That was me trying to be clever and caching the installation of rich-click. But it doesn't make any sense in other repos. I'll drop it now and do a patch release.

**Comment by ewels** [comment]
Fixed in f773ea04596b6cb53eebe1f1cadf3c96da7807e1 - will do a patch release now.

**Comment by ewels** [comment]
Ok you should be able to delete that file now. Thanks for reporting 👍🏻
